openrelik_worker_common/mount_utils.py

```diff
@@ -0,0 +1,309 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import subprocess
+from uuid import uuid4
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class BlockDevice:
+    """BlockDevice provides functionality to map image files to block devices
+    and mount them.
+
+    Usage:
+        bd = BlockDevice('/folder/path_to_disk_image.dd')
+        mountpoints = bd.mount()
+        # Do the things you need to do :)
+        bd.destroy()
+    """
+
+    MIN_PARTITION_SIZE = 100000000  # 100MB
+
+    def __init__(self, image_path: str, min_partition_size: int = MIN_PARTITION_SIZE):
+        """Initialize BlockDevice class instance.
+
+        Args:
+            image_path (str): path to the image file to map and mount.
+            min_partition_size (int): minimum partition size, default MIN_PARTITION_SIZE
```

Comment:
I would opt to make this magic number a bit more descriptive, something like:

```suggestion
class BlockDevice:
    """BlockDevice provides functionality to map image files to block devices
    and mount them.

    Usage:
        bd = BlockDevice('/folder/path_to_disk_image.dd')
        mountpoints = bd.mount()
        # Do the things you need to do :)
        bd.destroy()
    """

    MIN_PARTITION_SIZE_BYTES = 100 * 1024 * 1024  # 100 MB

    def __init__(self, image_path: str, min_partition_size: int = MIN_PARTITION_SIZE_BYTES):
        """Initialize BlockDevice class instance.

        Args:
            image_path (str): path to the image file to map and mount.
            min_partition_size (int): minimum partition size, default MIN_PARTITION_SIZE_BYTES
```

Reply:
agree, done.